Rewrite precedence rules for more clarify (#2112)
ansible/ansible-documentation: docs/docsite/rst/playbook_guide/playbooks_variables.rst

  #. command line values (for example, ``-u my_user``, these are not variables)
  #. role defaults (as defined in :ref:`Role directory structure <role_directory_structure>`) [1]_

>   #. variables for group "all" defined inside of an inventory file or script [2]_

Reviewer: lets use 'inventory source' ... also need a global definition of the term

  #. other group vars defined inside of an inventory file or script [2]_
  #. inventory group_vars/all (a file in the directory group_vars/ adjacent to the inventory) [3]_
  #. playbook group_vars/all (a file in the directory group_vars/ adjacent to the playbook) [3]_
  #. inventory group_vars/* (a file in the directory group_vars/ adjacent to the inventory) [3]_
  #. playbook group_vars/* (a file in the directory group_vars/ adjacent to the playbook) [3]_
  #. host vars defined in an inventory file or script [2]_
  #. inventory host_vars/* (a file in the directory host_vars/ adjacent to the inventory) [3]_

>   #. playbook host_vars/* (a file in the directory host_vars/ adjacent to the inventory) [3]_

Reviewer 2: ```suggestion
  #. inventory ``group_vars/all`` (a file in the ``group_vars/`` directory that is adjacent to the inventory) [3]_
  #. playbook ``group_vars/all`` (a file in the ``group_vars/`` directory that is adjacent to the playbook) [3]_
  #. inventory ``group_vars/*`` (a file in the ``group_vars/`` directory that is adjacent to the inventory) [3]_
  #. playbook ``group_vars/*`` (a file in the ``group_vars/`` directory that is adjacent to the playbook) [3]_
  #. host vars defined in an inventory file or script [2]_
  #. inventory ``host_vars/*`` (a file in the ``host_vars/`` directory that is adjacent to the inventory) [3]_
  #. playbook ``host_vars/*`` (a file in the ``host_vars/`` directory that is adjacent to the inventory) [3]_
```

  #. host facts / cached set_facts [4]_

>   #. play vars (defined in the section vars for the play)

Reviewer 2: ```suggestion
  #. play vars (defined in the ``vars`` section for the play)
```

Reviewer: there are also vars_prompt and vars_files

Author: It was added by Don Naro

Reviewer 3: I don't see such an addition. Adding this note only for `vars` is odd. `vars_prompt` and `vars_files` are listed on the lines below and haven't been modified in in your branch.

  #. include params
  #. extra vars (for example, ``-e "user=my_user"``)(always win precedence)

> If inventory file is located in the same directory as a playbook, adjacent group_vars/ are interpreted twice both as inventory group_vars/ and playbook group_vars/, therefore, getting precedence of the playbook group_vars.

Reviewer 2: ```suggestion
If an inventory file is located in the same directory as a playbook, adjacent ``group_vars/`` are interpreted twice, both as inventory ``group_vars/`` and playbook ``group_vars/``. As a result, the ``group_vars/`` from the inventory file get precedence over the playbook ``group_vars``.
```

Reviewer: This is incorrect, at no point does the inventory file get precedence over the playbook's group_vars.

What happens is that since the file is processed x2, it ends up with the highest precedence, that of the playbook relative file.

Author: Hello, thank you for suggestion I accept it, but correct for playbook been 'stronger'.

Author: It was added by Don Naro

Reviewer 3: The formatting and grammar are better in the suggested change. I don't think it's useful to mention that playbook group_vars get precedence over inventory group vars in this sentence though, they're the same.

```suggestion
If an inventory file is located in the same directory as a playbook, adjacent ``group_vars/`` are interpreted twice, both as inventory ``group_vars/`` and playbook ``group_vars/``.
```
